core/service.go

```diff
@@ -88,14 +88,34 @@ func NewService(cfg *Config) (*Service, error) {
 
 	epochDone := make(chan struct{})
 
+	srv := &Service{
+		rt:           cfg.Runtime,
+		keys:         keys,
+		blkRec:       cfg.NewBlocks, // becomes block receive channel in core service
+		msgRec:       cfg.MsgRec,
+		msgSend:      cfg.MsgSend,
+		blockState:   cfg.BlockState,
+		storageState: cfg.StorageState,
+		epochDone:    epochDone,
+	}
+
+	authData, err := srv.retrieveAuthorityData()
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: our authority index should be in authData, if it isn't, we aren't authorities
+	// need to add our authority data to storage
```

Comment:
Do we have an issue open to address this?

Reply:
I'll open one

Reply:
#598